Example/Backpack NativeUITests/CalendarUITest.swift

```diff
@@ -0,0 +1,88 @@
+
+class CalendarUITest: BackpackUITestCase {
+    private func goToPreselectedDatesCalendar(_ app: XCUIApplication) {
+        let tablesQuery = app.tables
+        tablesQuery.staticTexts["Calendar"].tap()
+        tablesQuery.staticTexts["With preselected dates"].tap()
+    }
+    
+    func testCalendarWithStartDateSelectedCanSelectDifferentDate() {
+        let app = XCUIApplication()
+        goToPreselectedDatesCalendar(app)
+        
+        XCTAssertTrue(app.cells["March 12, 2020, Selected as departure date"].isSelected)
+        XCTAssertTrue(app.cells["March 20, 2020, Selected as return date"].isSelected)
+        
+        let collectionViewsQuery = app.collectionViews
+        collectionViewsQuery.cells["March 10, 2020"].staticTexts["10"].tap()
+        collectionViewsQuery.cells["March 24, 2020"].staticTexts["24"].tap()
+        
+        XCTAssertTrue(app.cells["March 10, 2020, Selected as departure date"].isSelected)
+        XCTAssertTrue(app.cells["March 24, 2020, Selected as return date"].isSelected)
+        
+        XCTAssertFalse(app.cells["March 12, 2020, Between departure and return date"].isSelected)
+        XCTAssertFalse(app.cells["March 20, 2020, Between departure and return date"].isSelected)
+    }
+
+    func testCanSelectDifferentDatesOnSingleSelectionCalendar() {
+        let app = XCUIApplication()
+        goToPreselectedDatesCalendar(app)
+        
+        app.buttons["Single"].tap()
+        
+        let collectionViewsQuery = app.collectionViews
+        
+        collectionViewsQuery.cells["March 1, 2020"].staticTexts["1"].tap()
+        XCTAssertTrue(app.cells["March 1, 2020"].isSelected)
+        XCTAssertFalse(app.cells["April 1, 2020"].isSelected)
+        
+        collectionViewsQuery.cells["April 1, 2020"].staticTexts["1"].tap()
+        XCTAssertTrue(app.cells["April 1, 2020"].isSelected)
+        XCTAssertFalse(app.cells["March 1, 2020"].isSelected)
+    }
+    
+    func testCanSelectMultipleDatesOnMultiSelectionCalendar() {
+        let app = XCUIApplication()
+        goToPreselectedDatesCalendar(app)
+        
+        app.buttons["Multiple"].tap()
+        
+        XCTAssertFalse(app.cells["March 1, 2020"].isSelected)
+        XCTAssertFalse(app.cells["March 3, 2020"].isSelected)
+        XCTAssertFalse(app.cells["March 16, 2020"].isSelected)
+        XCTAssertFalse(app.cells["March 10, 2020"].isSelected)
+        XCTAssertFalse(app.cells["March 13, 2020"].isSelected)
+        
+        let collectionViewsQuery = app.collectionViews
+        collectionViewsQuery.cells["March 1, 2020"].staticTexts["1"].tap()
+        collectionViewsQuery.cells["March 3, 2020"].staticTexts["3"].tap()
+        collectionViewsQuery.cells["March 16, 2020"].staticTexts["16"].tap()
+        collectionViewsQuery.cells["March 10, 2020"].staticTexts["10"].tap()
+        collectionViewsQuery.cells["March 13, 2020"].staticTexts["13"].tap()
+        
+        XCTAssertTrue(app.cells["March 1, 2020"].isSelected)
```

Comment:
should we also include other dates that are NOT selected? (ie not the days on either side of a date for example)

Reply:
Yes! But it would be hard to know if the tests is actually effective.
Since picking the unselected dates to check would be discretional, we may never see the issue (if one comes out)
An option would be rather brute-force checking of all other dates. WDYT?

Reply:
Yeah! I can add those checks!
Although I'm not really sure if it would add any value, since to be sure nothing else is selected, we should check all other dates. This would fit in a Unit Tests rather than a UI one I think.
WDYT?

Reply:
yeah i guess the alternative is that it's similar to the above test where some get selected & then unselected